### PR TITLE
Clang tidy (fix/suppress)

### PR DESCRIPTION
**Pull request by ruslo**

### CODE REVIEW

File: src/app/landmarks/DlibXML.h [hunk 1]
```diff
@@ -72,8 +72,8 @@ inline cv::Size read_png_size(const std::string& filename)
     {
         unsigned int width, height;
         in.seekg(16);
-        in.read(reinterpret_cast<char*>(&width), 4);
-        in.read(reinterpret_cast<char*>(&height), 4);
+        in.read(reinterpret_cast<char*>(&width), 4); // NOLINT (TODO)
+        in.read(reinterpret_cast<char*>(&height), 4); // NOLINT (TODO)
```
Comment (headupinclouds) [inline src/app/landmarks/DlibXML.h]: So basically every `reinterpret_cast<>` is flagged by `clang-tidy` even if the interface may require it in some cases http://www.cplusplus.com/reference/istream/istream/read/ `istream& read (char* s, streamsize n);`?  Does `clang-tidy` suggest an alternative?

File: src/lib/drishti/core/Field.h [hunk 2]
```diff
@@ -24,16 +24,20 @@ DRISHTI_CORE_NAMESPACE_BEGIN
 template <typename T>
 struct Field
 {
-    Field()
-         
-    = default;
+    Field() = default;
```
Comment (headupinclouds) [inline src/lib/drishti/core/Field.h]: Eventually we can use [optiona](https://en.cppreference.com/w/cpp/utility/optional) for this (C++17).

File: src/lib/drishti/core/Logger.cpp [hunk 3]
```diff
@@ -42,7 +42,7 @@ std::shared_ptr<spdlog::logger> Logger::create(const char* name)
     {
         // Check for existing logger
         auto logger = get(name);
-        if (logger)
+        if (logger) // NOLINT (TODO)
```
Comment (headupinclouds) [inline src/lib/drishti/core/Logger.cpp]: I'm also curious what `clang-tidy` would recommend here.

File: src/lib/drishti/drishti/EyeIO.cpp [hunk 4]
```diff
@@ -60,7 +60,7 @@ void serialize(Archive& ar, drishti::sdk::Size2i& s, const unsigned int version)
 template <class Archive>
 void serialize(Archive& ar, drishti::sdk::Size2f& s, const unsigned int version)
 {
-    ar& GENERIC_NVP("width", s.width);
+    ar& GENERIC_NVP("width", s.width); // NOLINT (TODO)
```
Comment (headupinclouds) [inline src/lib/drishti/drishti/EyeIO.cpp]: Alternative?

**Comment by headupinclouds** [comment]
Added https://github.com/elucideye/drishti/issues/716

**Comment by ruslo** [comment]
> explicit constructors in derived classes: explicit VideoSourceStills(const std::string& filename)

> no extra whitespace

Blame me for these ones :)

**Pull request merged**